src/tezos_interop/tezos_bridge.ml

```diff
@@ -2,6 +2,36 @@ open Helpers
 open Crypto
 open Tezos
 
+module Michelson = struct
+  include Michelson
+  type t = Michelson.t
+  let of_yojson json =
+    let%ok json = Yojson.Safe.to_string json |> Data_encoding.Json.from_string in
+    try Ok (Data_encoding.Json.destruct Michelson.expr_encoding json) with
+    | _ -> Error "invalid json"
+end
+module Listen_transaction = struct
+  type kind = Listen
```

Comment:
I am not sure about this, but I feel like this `kind = listen` is kinda the same that the one in `Pending`.
Do they have the same functional meaning?

Reply:
@gsebil08 yes and they're the same as the ones expected by the JavaScript part (see tezos_js_bridge.js).

@EduardoRFS: Following @gsebil08 question, is there a reason you went for a polymorphic `request` function with explicit parameters to handle conversion instead of using modules/objects?

Reply:
@aguillon 
>yes and they're the same as the ones expected by the JavaScript part (see tezos_js_bridge.js).

They **have the same functional meaning**, but they **are not** the same :)
That was the hidden part of my question, why don't we define it only once?

Reply:
Sorry, where is kind defined twice? Pending.kind is not the same as Listen_transaction.kind, they don't have the same functional meaning either.

If your question is `why not define as Listen | Pending` because that would allow to describe more bad states.